### ISSUE TITLE
more helpful css classes for rounded corners

### ISSUE THREAD

**Issue by coffeeandwork** [beercss/beercss#170]
At present Beer CSS has these css classes that help specify how rounded the corners of UI elements are:
- `square` or `no-round` when you want sharp corners
- `small-round` when you want corners to be slightly rounded
- `round` or `medium-round` when you want corners to have more rounding
- `large-round` when you want the most rounded corners available via Beer CSS.

And then there are these css classes that allow you to conveniently specify which of the 4 corners should be rounded:
- `left-round`
- `right-round`
- `top-round`
- `bottom-round`

It would be useful if Beer CSS combined the above two ideas and provided classes that let you choose which corners to round as well as how much to round so that there are classes such as these below:
- `left-<size>-round`
- `right-<size>-round`
- `top-<size>-round`
- `bottom-<size>-round`

In the above class names we can replace `<size>` with either `small`, `medium`, or `large`. 

Thoughts?

**Comment by leonardorafael** [comment]
Hi @coffeeandwork ,

Instead this:

```html
<div class="left-small-round top-small-round"></div>
```

I would like to suggest this:

```html
<div class="small-round left-round top-round"></div>
```

Doing that, maybe we can maintain the bundle size of project below 10kb, using some local css variables.

**Comment by coffeeandwork** [comment]
> Hi @coffeeandwork ,
> 
> Instead this:
> 
> ```
> <div class="left-small-round top-small-round"></div>
> ```
> 
> I would like to suggest this:
> 
> ```
> <div class="small-round left-round top-round"></div>
> ```
> 
> Doing that, maybe we can maintain the bundle size of project below 10kb, using some local css variables.

That sounds good. Thank you.

**Comment by coffeeandwork** [comment]
If we can shorten "small" to "s", "medium" to "m", "large" to "l", it may help further reduce bundle size without compromising much of developer friendliness since s, m, and, l are already used elsewhere in the library to indicate sizes.

**Comment by KicKerBNU** [comment]
Its a good idea but i personally dont like this approach of s m l or any other one word classes, the class name should stay readable  for people who never used beercss.

**Comment by coffeeandwork** [comment]
What about "sm", "lg", "md"? Maybe that's a good compromise between small size and readability?

**Comment by laurentpayot** [comment]
> What about "sm", "lg", "md"? Maybe that's a good compromise between small size and readability?

For me that would be more confusing than just `s`, `l` and `m`.
My first thought when reading `sm` is "small to medium".
My first thought when reading `md` is "markdown" :laughing:.

**Comment by coffeeandwork** [comment]
> > What about "sm", "lg", "md"? Maybe that's a good compromise between small size and readability?
> 
> For me that would be more confusing than just `s`, `l` and `m` (which are already used for responsiveness anyway). My first thought when reading `sm` is "small to medium". My first thought when reading `md` is "markdown" 😆.

Here are some data points regarding what other frameworks seem to be doing:

Frameworks that use sm, md, lg:
- Bootstrap: https://getbootstrap.com/docs/5.3/layout/grid/#grid-options. 
- tailwind: https://tailwindcss.com/docs/responsive-design
- MUI: https://mui.com/material-ui/react-grid/#how-it-works
- mdbootstrap: https://mdbootstrap.com/docs/standard/layout/flexbox/

Frameworks that use the fully spelled out small, medium, large:
- Sales force lightening design system: https://www.lightningdesignsystem.com/utilities/grid/#Responsive-Layout-Options

Frameworks that don't use small, medium, large at all, and instead opt for using more specific naming like tablet, desktop, mobile, etc:
- US gov't: https://designsystem.digital.gov/utilities/layout-grid/#responsive-variants-2
- UK gov't: https://design-system.service.gov.uk/styles/layout/#grid-system


Bootstrap, tailwind, material design (MUI), are by far the most widely used web frameworks in the world today, afaik. Developers who work with those frameworks will have seen a lot of sm/md/lg, and a lot less of the other variations in naming. So if we go with sm/md/lg, then we will be ensuring the naming of these css classes are familiar to the largest number of developers today. Only reason to deviate from sm/md/lg to something else would be because of some other compelling reason, not possible confusion/misinterpretation.

I am totally fine with whichever direction Beer CSS goes. I hope this info is helpful in making the decision.

**Comment by laurentpayot** [comment]
I did a test and in the current 59376 bytes beer.min.css I replaced 56 occurences of `small` by `sm`, 46 occurences of `medium` by `md`, and 62 occurences of `large` by `lg`. It created a 58838 bytes new file (less than 1% smaller).
With brotli compression (`brotli -Zcn beer.min.css | wc -c`) the initial file was 9872 bytes, the new one 9815 (57 bytes difference, less than 1% smaller).

As the code is read at least 10 times more than it is written, I think that in terms of developer experience the readability of the fully spelled words is worth 57 bytes of transferred data.
